x-pack/legacy/plugins/monitoring/public/components/alerts/configuration/configuration.tsx

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ReactNode } from 'react';
+import { kfetch } from 'ui/kfetch';
+import { EuiSteps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ActionResult } from '../../../../../actions/server/types';
+import { ALERT_ACTION_TYPE_EMAIL } from '../../../../common/constants';
+import { getMissingFieldErrors } from '../../../lib/form_validation';
+import { Step1 } from './step1';
+import { Step2 } from './step2';
+import { Step3 } from './step3';
+
+export interface AlertsConfigurationProps {
+  emailAddress: string;
+  onDone: Function;
+}
+
+export interface StepResult {
+  title: string;
+  children: ReactNode;
+  status: any;
+}
+
+export interface AlertsConfigurationForm {
+  email: string | null;
+}
+
+export const NEW_ACTION_ID = '__new__';
+
+export const AlertsConfiguration: React.FC<AlertsConfigurationProps> = (
+  props: AlertsConfigurationProps
+) => {
+  const { onDone } = props;
+
+  const [emailActions, setEmailActions] = React.useState<ActionResult[]>([]);
+  const [selectedEmailActionId, setSelectedEmailActionId] = React.useState('');
+  const [editAction, setEditAction] = React.useState<ActionResult | null>(null);
+  const [emailAddress, setEmailAddress] = React.useState(props.emailAddress);
+  const [formErrors, setFormErrors] = React.useState<AlertsConfigurationForm>({ email: null });
+  const [showFormErrors, setShowFormErrors] = React.useState(false);
+  const [isSaving, setIsSaving] = React.useState(false);
+  const [saveError, setSaveError] = React.useState('');
+
+  React.useEffect(() => {
+    async function fetchData() {
+      await fetchEmailActions();
+    }
+
+    fetchData();
+  }, []);
+
+  React.useEffect(() => {
+    setFormErrors(getMissingFieldErrors({ email: emailAddress }, { email: '' }));
+  }, [emailAddress]);
```

Comment:
Seems kinda strange to have two `useEffect` in one component. Maybe you can do something like...
```javascript
React.useEffect(async () => {
    await fetchEmailActions();
    setFormErrors(getMissingFieldErrors({ email: emailAddress }, { email: '' }));
}, [emailAddress]);
```

Reply:
[I think the React team recommends this](https://reactjs.org/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns)

Reply:
Indeed! Thanks for schooling me 🙇